st3/lsp_utils/_client_handler/abstract_plugin.py

```diff
@@ -128,12 +130,16 @@ def can_start(cls, window: sublime.Window, initiating_view: sublime.View,
     @classmethod
     def on_pre_start(cls, window: sublime.Window, initiating_view: sublime.View,
                      workspace_folders: List[WorkspaceFolder], configuration: ClientConfig) -> Optional[str]:
-        extra_paths = path.pathsep.join(cls.get_additional_paths())
+        # To fix https://github.com/TerminalFi/LSP-copilot/issues/163 ,
+        # We don't want to add the same path multiple times whenever a new server session is created.
+        extra_paths = cls.get_additional_paths()
         if extra_paths:
-            original_path = configuration.env.get('PATH') or ''
-            if isinstance(original_path, list):
-                original_path = path.pathsep.join(original_path)
-            configuration.env['PATH'] = path.pathsep.join([extra_paths, original_path])
+            original_path_raw = configuration.env.get('PATH') or ''
+            if isinstance(original_path_raw, str):
+                original_paths = original_path_raw.split(path.pathsep)
+            else:
+                original_paths = original_path_raw
+            configuration.env['PATH'] = path.pathsep.join(unique_everseen(chain(extra_paths, original_paths)))
```

Comment:
We just need an `if ... in` check. No need for over-engineering with some function that will clean up unrelated paths.

And also it's even potentially wrong since it's using set which doesn't guarantee order to be the same as the input order.

Reply:
> We just need an if ... in check. No need for over-engineering with some function that will clean up unrelated paths.

Fine. Will do.

> And also it's even potentially wrong since it's using set which doesn't guarantee order to be the same as the input order.

No. It's `set` is only for checking seen or not seen. The order is maintained.